init.lua

```diff
@@ -842,6 +842,51 @@ function Form:set_as_inventory_for(player, ctx)
     player:set_inventory_formspec(fs)
 end
 
+-- Declared here to be accessible by render_to_formspec_string
+local fs_process_events
+
+-- Prevent collisions in forms, but also ensure they don't happen across
+-- mutliple embedded forms within a single parent.
+-- Unique per-user to prevent players from making the counter wrap around for
+-- other players.
+local render_to_formspec_auto_name_ids = {}
+-- if `embedded` is true, then this won't output a "root" formspec, but instead
+--   will output  formspec designed to be embedded along with the information
+--   that was removed from the form
+function Form:render_to_formspec_string(player, ctx, embedded)
+    local name = player:get_player_name()
+    local info = minetest.get_player_information(name)
+    local tree, form_info = self:_render(player, ctx or {},
+        info and info.formspec_version, render_to_formspec_auto_name_ids[name])
+    local public_form_info
+    if embedded then
+        local size = table.remove(tree, 1)
+        public_form_info = {w = size.w, h = size.h,
+            formspec_version = tree.formspec_version}
+        tree.formspec_version = nil
+    end
```

Comment:
```suggestion
-- If standalone is set, this will return a standalone formspec, otherwise it
-- will return a formspec that can be embedded and a table with its size and
-- target formspec version
function Form:render_to_formspec_string(player, ctx, standalone)
    local name = player:get_player_name()
    local info = minetest.get_player_information(name)
    local tree, form_info = self:_render(player, ctx or {},
        info and info.formspec_version, render_to_formspec_auto_name_ids[name])
    local public_form_info
    if not standalone then
        local size = table.remove(tree, 1)
        public_form_info = {w = size.w, h = size.h,
            formspec_version = tree.formspec_version}
        tree.formspec_version = nil
    end
```

Reply:
sorry, hadn't seen this review